starcounter-include.html

```diff
@@ -238,22 +248,43 @@
          */
         StarcounterIncludePrototype._compositionChanged = function(compositionString) {
             if (compositionString) {
-                this._forceLayoutChange = true;
-                this._renderCompositionChange("explicit or fallback", () => this.stringToDocumentFragment(compositionString));
+                this.explicitComposition = compositionString;
+                return this._renderCompositionChange(compositionString, () => this.stringToDocumentFragment(compositionString));
             }
             else {
                 const compositionProvider = this.partial && findCompositionProvider(this.partial);
 
                 if (compositionProvider) {
                     this.partialId = compositionProvider.PartialId;
+                    this.explicitComposition = null;
                     if (compositionString === "") {
                         //this is a request from starcounter-layout-html-editor to reset to default composition
-                        this.Composition$ = "";
                     }
                     else if (compositionProvider.Composition$) {
                         return this._renderCompositionChange(compositionProvider.Composition$, () => {
                             this.storedLayout = compositionProvider.Composition$; //should always be string
-                            return this.stringToDocumentFragment(this.storedLayout);
+                            const merged = this.stringToDocumentFragment(this.storedLayout);
+
+                            if (compositionProvider.ViewUris) {
+                                const keys = Object.keys(this.partial);
+                                for(let key of keys) {
+                                    const scoped = this.partial[key];
+                                    if (scoped && scoped.Html && compositionProvider.ViewUris.indexOf(scoped.Html) === -1) {
+                                        if (this.defaultComposition) {
+                                            this.template.scopedNodes.forEach((nodes) => {
+                                                if(nodes.scope === key) {
+                                                    nodes.forEach((node) => appendComposition(merged, node));
+                                                }
+                                            });
+                                        }
+                                        else {
+                                            return false;
```

Comment:
If I get it riht, we will end in this `else` if 
- we have specific composition
- we have some views that are not covered by specific composition, 
- we do not have any defualt composition.

Reply:
Yeah, I think you are right. Can you fix this and add a test?

Reply:
https://github.com/Starcounter/starcounter-include/issues/72